docs(asset,assetcard,avatar,badge): site docs to storybook

mdt2 opened the pull request:
## Description

This migrates the documentation from the deprecated docs site over to Storybook, including showing the variants on the Storybook Docs page.

- Asset - got a new MDX page and some MDX-only stories
- Assetcard - got a new MDX page but was able to use existing stories
- Assetlist - no changes required since this one already matches the docs site as is
- Avatar - got a new MDX page and some MDX-only stories
- Badge - got a new MDX page and some MDX-only stories

This PR doesn't need a changeset since it's docs-only.

## How and where has this been tested?

Please tag yourself on the tests you've marked complete to confirm the tests have been run by someone other than the author.

### Validation steps

- [x] The [Asset storybook docs](https://pr-2782--spectrum-css.netlify.app/preview/?path=/docs/components-asset--docs) include all necessary info when compared to the [docs site](https://pr-2782--spectrum-css.netlify.app/asset)
- [x] The [Assetcard storybook docs](https://pr-2782--spectrum-css.netlify.app/preview/?path=/docs/components-asset-card--docs) include all necessary info when compared to the [docs site](https://pr-2782--spectrum-css.netlify.app/assetcard)
- [x] The [Avatar storybook docs](https://pr-2782--spectrum-css.netlify.app/preview/?path=/docs/components-avatar--docs) include all necessary info when compared to the [docs site](https://pr-2782--spectrum-css.netlify.app/avatar)
- [x] The [Bade storybook docs](https://pr-2782--spectrum-css.netlify.app/preview/?path=/docs/components-badge--docs) include all necessary info when compared to the [docs site](https://pr-2782--spectrum-css.netlify.app/badge)

### Regression testing

Validate:

1. The documentation pages for at least two other components are still loading, including:

- [ ] The pages render correctly, are accessible, and are responsive.

2. If components have been modified, VRTs have been run on this branch:

- [ ] VRTs have been run and looked at.
- [ ] Any VRT changes have been accepted (by reviewer and/or PR author), or there are no changes.


## To-do list

- [x] I have read the [contribution guidelines](/.github/CONTRIBUTING.md).
- [x] I have updated relevant storybook stories and templates.
- [x] If my change impacts **documentation**, I have updated the documentation accordingly.
- [ ] ✨ This pull request is ready to merge. ✨

rise-erpelding reviewed (commented):
Just like the last one, this one looks pretty solid! I just noted one potential typo above and had questions about:

* **Avatar**: [The Migration Guide section](https://opensource.adobe.com/spectrum-css/avatar.html#migrationguide) where it talks about needing a div wrapper, was there a specific reason this wasn't carried over to Storybook? I saw that with Badge you did include the detail about the label and icon elements needing to be nested inside the parent container `.spectrum-Badge` and this bit on Avatar seemed similar.
* **Badge**: There's no sizing story for this one like on the docs site and I have a feeling that has something to do with us letting controls/Chromatic show that instead, but want to confirm. By the way, I really liked how you condensed the information for Badge and were able to include text wrapping and content (label only, icon only, icon with label etc.) within the default story.

mdt2 commented:
Thanks for this feedback @rise-erpelding 🎉 

- **Avatar** - great catch, I've added this in! I also added the list of available sizing classes from the migration guide to the Sizing example
- **Badge** - this is a great question and your assumption is correct. Since almost all the components have t-shirt sizing I was leaving that out. I'm definitely open to adjustments on this though if we decide we want to include it. I think it probably makes sense to include for components outside of the normal t-shirt sizing (the sizing in Avatar for instance)

rise-erpelding reviewed (approved):
All of the previous feedback items have been addressed! I took one more quick look at all of the components involved here as well and I think this one is good to go!
